crates/bevy_asset/src/io/memory.rs

```diff
@@ -361,6 +409,183 @@ impl AssetReader for MemoryAssetReader {
     }
 }
 
+/// A writer that writes into [`Dir`], buffering internally until flushed/closed.
+struct DataWriter {
+    /// The dir to write to.
+    dir: Dir,
+    /// The path to write to.
+    path: PathBuf,
+    /// The current buffer of data.
+    ///
+    /// This will include data that has been flushed already.
+    current_data: Vec<u8>,
+    /// Whether to write to the data or to the meta.
+    write_meta: bool,
```

Comment:
Might make sense to use an enum here instead of a bool

Reply:
alternatively, rename it to `is_meta_writer`

Reply:
Done.